File: app/src/main/java/art/luaj/hyperisle/ui/view/preference/MaterialSwitchPreference.java
package art.luaj.hyperisle.ui.view.preference;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.preference.SwitchPreferenceCompat;

import art.luaj.hyperisle.R;

public class MaterialSwitchPreference extends SwitchPreferenceCompat {
    public MaterialSwitchPreference(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init();
    }

    public MaterialSwitchPreference(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    public MaterialSwitchPreference(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public MaterialSwitchPreference(@NonNull Context context) {
        super(context);
        init();
    }

    private void init() {
        setWidgetLayoutResource(R.layout.pref_m3_switch);
    }
}

File: app/src/main/java/art/luaj/hyperisle/ui/view/MaterialSwitchPreference.kt
package art.luaj.hyperisle.ui.view

import android.content.Context
import android.util.AttributeSet
import androidx.preference.SwitchPreferenceCompat
import art.luaj.hyperisle.R

class MaterialSwitchPreference : SwitchPreferenceCompat {
    init {
        widgetLayoutResource = R.layout.pref_switch
    }

    constructor(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        defStyleRes: Int
    ) : super(context, attrs, defStyleAttr, defStyleRes)

    constructor(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int
    ) : super(context, attrs, defStyleAttr)

    constructor(
        context: Context,
        attrs: AttributeSet?
    ) : super(context, attrs)

    constructor(context: Context) : super(context)
}